homeassistant/components/simplisafe/camera.py

```diff
@@ -0,0 +1,243 @@
+            """Executor helper to write image."""
+            os.makedirs(os.path.dirname(to_file), exist_ok=True)
+            with open(to_file, "wb") as img_file:
+                img_file.write(image_data)
+
+        async def save_image_handler(call: ServiceCall) -> None:
+            if self._attr_image_url is None:
+                return
+
+            width = call.data.get(ATTR_WIDTH)
+            filename: Template = call.data.get(ATTR_FILENAME)  # type: ignore[assignment]
+            filename.hass = self._hass
+            snapshot_file = filename.async_render(variables={ATTR_ENTITY_ID: self})
+
+            try:
+                snapshot = await self._simplisafe.async_media_request(
+                    self._attr_image_url.replace("{&width}", "&width=" + str(width))
+                )
+            except SimplipyError as err:
+                raise HomeAssistantError(
+                    f'Error fetching motion media "{self._system.system_id}": {err}'
+                ) from err
+
+            try:
+                await self._hass.async_add_executor_job(
+                    _write_image, snapshot_file, snapshot
+                )
+            except OSError as err:
+                LOGGER.error("Can't write image to file: %s", err)
+
+        async def save_clip_handler(call: ServiceCall) -> None:
+            if self._attr_clip_url is None:
+                return
+
+            filename: Template = call.data.get(ATTR_FILENAME)  # type: ignore[assignment]
+            filename.hass = self._hass
+            clip_file = filename.async_render(variables={ATTR_ENTITY_ID: self})
+
+            try:
+                clip = await self._simplisafe.async_media_request(self._attr_clip_url)
+            except SimplipyError as err:
+                raise HomeAssistantError(
+                    f'Error fetching motion media "{self._system.system_id}": {err}'
+                ) from err
+
+            try:
+                await self._hass.async_add_executor_job(_write_image, clip_file, clip)
+            except OSError as err:
+                LOGGER.error("Can't write image to file: %s", err)
+
+        self._hass.services.async_register(
+            DOMAIN,
+            SERVICE_OC_IMAGE,
+            save_image_handler,
+            schema=SERVICE_OC_IMAGE_SCHEMA,
+        )
+
+        self._hass.services.async_register(
+            DOMAIN,
+            SERVICE_OC_CLIP,
+            save_clip_handler,
+            schema=SERVICE_OC_CLIP_SCHEMA,
+        )
+
+    @callback
+    def async_update_from_websocket_event(self, event: WebsocketEvent) -> None:
+        """Receive a Simplisafe WebsocketEvent aimed at me specifically, saving the media urls for later."""
+        if event.event_type != EVENT_CAMERA_MOTION_DETECTED:
+            return
+        if event.sensor_type != DeviceTypes.OUTDOOR_CAMERA:
+            return
+
+        if event.media_urls is None:
+            LOGGER.error("MISSING MEDIA URLS!")
```

Comment:
Should this really be an error? The user can't do anything to correct it.

Reply:
Ok, removed error printing